packages/web-components/src/components/va-date-text-input/va-date-text-input.tsx

```diff
@@ -0,0 +1,256 @@
+import { Component, Event, EventEmitter, Host, Prop, h } from '@stencil/core';
+
+import { days, isFullDate } from '../../utils/date-utils';
+@Component({
+  tag: 'va-date-text-input',
+  styleUrl: 'va-date-text-input.css',
+  shadow: true,
+})
+export class VaDateTextInput {
+  /**
+   * Render marker indicating field is required.
+   */
+  @Prop() required: boolean;
+
+  /**
+   * Label for the field.
+   */
+  @Prop() label!: string;
+
+  /**
+   * Used to create unique name attributes for each input.
+   */
+  @Prop() name!: string;
+
+  /**
+   * The error message to render (if any)
+   * This prop should be leveraged to display any custom validations needed for this component
+   */
+  @Prop() error: string;
+
+  /**
+   * The aria-describedby attribute lists the ids of the elements that describe the object. 
+   * It is used to establish a relationship between widgets or groups and the text that describes them
+   */
+  @Prop() ariaDescribedby: string;
+
+  /**
+   * Set to true if custom validation logic does not meet expected criteria for valid input
+   */
+  @Prop() customValidationBoolean: boolean;
+
+  /**
+   * Set message to display if custom validation boolean is true
+   */
+  @Prop() customValidationMessage: string;
+
+  /**
+   * Set the default date value must be in YYYY-MM-DD format.
+   */
+  @Prop({ mutable: true }) value: string;
+
+  /**
+   * Fires when the date input loses focus after its value was changed
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  dateChange: EventEmitter;
+
+  /**
+   * Fires when the date input loses focus
+   */
+  @Event({
+    composed: true,
+    bubbles: true,
+  })
+  dateBlur: EventEmitter;
+
+  private handleDateBlur = (event: FocusEvent) => {
+    const [year, month, day] = (this.value || '').split('-').map(val => val);
+    const yearNum = parseInt(year);
+    const monthNum = parseInt(month);
+    const dayNum = parseInt(day);
+    // Use a leading zero for numbers < 10
+    const numFormatter = new Intl.NumberFormat('en-US', {
+      minimumIntegerDigits: 2,
+    });
+    this.value = `${year}-${month ? numFormatter.format(monthNum) : ''}-${
+      day ? numFormatter.format(dayNum) : ''
+    }`;
+
+    const daysForSelectedMonth = monthNum > 0 ? days[monthNum] : [];
```

Comment:
I feel like we could add some checks here for leap years and only allow 29 days in February then, instead of always.

Reply:
@Mottie I have added a utility function to check for leap years and throw an error message if the date is invalid ie: February 29 1999 is invalid but February 29 2000 is valid